### ISSUE TITLE
Python 3.11 crashes in some cases when loguru.logger.exception called

### ISSUE THREAD

**Issue by gabus** [Delgan/loguru#775]
I'm sorry as I can't provide example but after digging into code the only change which is needed 

`formatter_record["exception"] = "".join(lines)` 
change to 
`formatter_record["exception"] = "".join(str(lines))`
in `loguru._handler.Handler.emit`

it seems python 3.11 doesn't like implicit casting. 
Though I'm second guessing my solution - don't know why it works..

**Comment by Delgan** [comment]
Thank you for the report. If you ever find out which exception reproduces the problem, let me know. I know there were different changes in exception handling in Python 3.11 which are not yet unit tested by Loguru, I'll try to find out what can cause this.

**Comment by gabus** [comment]
Managed to create a snippet which breaks

```
from loguru import logger
import falcon


class Ping:
    def on_get(self, req, resp):
        raise falcon.HTTPUnauthorized("test", "test unauthorised")


class ErrorHandler:
    def handle(self, req, resp, ex, params):
        logger.debug(ex)                   # works
        logger.exception(ex)               # dies here


def create_application():
    app = falcon.App()
    app.add_route('/ping', Ping())
    app.add_error_handler(falcon.HTTPError, ErrorHandler().handle)
    return app


application = create_application()
```

To build and run use
```
python3.11 -m venv venv
source venv/bin/activate
pip install pipenv

pipenv install loguru
pipenv install gunicorn
pipenv install falcon

venv/bin/gunicorn -p 8000 --reload --pythonpath ./ test_loguru
```

**Comment by Delgan** [comment]
Thanks for the snippet, @gabus.

Unfortunately, it seems I'm unable to reproduce the issue.
Which versions of `gunicorn` and `falcon` are you using?
What is the observed output of the code you shared?

**Comment by gabus** [comment]
Interesting!
Can it be anything to do with macOS python version? I'm on Monterey 12.6 mac with M1 chip. 

This is my setup:
```
Python 3.11.1

gunicorn==20.1.0
falcon==3.1.1
loguru==0.6.0
pipenv==2022.12.19
virtualenv==20.17.1
```

Output is very vague. It just shows that worker died and new one has been spawned. Then in separate pop up python crash window appears asking action: Ignore, Restart or Close Pyhton.
```
[2022-12-22 13:02:20 +0000] [53154] [WARNING] Worker with pid 53323 was terminated due to signal 11
[2022-12-22 13:02:20 +0000] [53339] [INFO] Booting worker with pid: 53339
```

**Comment by xzmeng** [comment]
I can reproduce it on macOS 13.1 with an M1 chip.

The process terminted with a segmentation violation when reading `tb.tb_frame`.

https://github.com/Delgan/loguru/blob/16dbf79b861bc4b7930c2915ee38794c9d3d2b1d/loguru/_better_exceptions.py#L182

**Comment by Delgan** [comment]
Thanks @gabus  and @xzmeng for the additional details.

I tested on Linux but have no access to MacOS computer unfortunately.

Are you both using Cython?
This recent pull request in `python/cpython` fixes a segfault in Python 3.11 when accessing `frame.f_back`, it does mention Cython as a environment where the bug happened: https://github.com/python/cpython/pull/100182
I think this is what you faced @xzmeng.
